tests/test_infrastructure_cache.py

```
@@ -0,0 +1,101 @@
+        key = KeyBuilder.build_sensor_key("loc123")
+        assert key == "sensors:loc123"
+
+    def test_measurement_key(self):
+        key = KeyBuilder.build_measurement_key("sensor456", page=5)
+        assert key == "measurements:sensor456:page:5"
+
+    def test_hash_key_consistency(self):
+        data = {"a": 1, "b": 2, "c": 3}
+        key1 = KeyBuilder.build_hash_key(data)
+        key2 = KeyBuilder.build_hash_key({"c": 3, "a": 1, "b": 2})
+        assert key1 == key2
```

Pylint (3.3.7)

[convention] Final newline missing

(C0304)